From e3e4a9cf57d5d7da6914b31527188eff7e290238 Mon Sep 17 00:00:00 2001
From: Feng Pan <fpan@redhat.com>
Date: Wed, 17 May 2017 11:39:31 -0400
Subject: [PATCH] Make sure overlay_ip_version is treated as string

puppet throws error if overlay_ip_version is passed in through hiera
as an integer. We should make sure this variable is treated as string
for the 'in' operator.

Closes-Bug: #1691502

Change-Id: I1c034e8018c959672b106e2d57992cf93b945d8a
Signed-off-by: Feng Pan <fpan@redhat.com>
---

diff --git a/manifests/plugins/ml2.pp b/manifests/plugins/ml2.pp
index d931e72..b943593 100644
--- a/manifests/plugins/ml2.pp
+++ b/manifests/plugins/ml2.pp
@@ -171,7 +171,7 @@
     warning ('supported_pci_vendor_devs is deprecated, has no effect and will be removed in a future release.')
   }
 
-  if !is_service_default($overlay_ip_version) and !($overlay_ip_version in [4, 6]) {
+  if !is_service_default($overlay_ip_version) and !("${overlay_ip_version}" in ['4', '6']) {
     fail('Invalid IP version for overlay_ip_version')
   }
 
